### DISCUSSION TITLE
Ereader import improvements

### DISCUSSION THREAD

**Discussion by Ceynou** [FreeLanguageTools/vocabsieve#40] [Ideas]
This is the [script](https://github.com/Kartoffel0/Kindle2Anki) I use to imports JP lookups.
The way it works is:
- You put the vocab.db file in the same directory.
- you feed it some yomichan glossary and frequency dictionaries zip, it saves them to one json.
- you feed it with your note type information and the field you want to fill, as well as some options, it saves those to a json file.
- Then it prompts you with a list of books you want to exports lookups from which you select by entering its number, then the number of words to extract
- then it asks you if you want to save the "last mined from" date
- also saves a list of words you extracted so you don't end up with duplicates next times

It can probably be improved with less prompts as well, I think the **last 2** features are what vocabsieve needs

**Comment by 1over137** [comment]
The new version should be able to store the last import date, and it will also set the start date appropriately next time you import.
I mean, Anki doesn't allow you to have duplicated notes, so I think the last one is a non-issue, even though there might be overlaps (the notes on the day of last import will be re-imported), because it will just fail.

**Reply by Ceynou** [reply]
Thanks!
Also I think the recent anki versions allow duplicates
